local-volume/provisioner/pkg/deleter/deleter.go

```diff
@@ -126,69 +126,53 @@ func (d *Deleter) asyncDeleteFilePV(pv *v1.PersistentVolume, mountPath string, c
 		d.RuntimeConfig.Recorder.Eventf(pv, v1.EventTypeWarning, common.EventVolumeFailedDelete,
 			deletingLocalPVErr.Error())
 		glog.Error(deletingLocalPVErr)
-		return err
+		return
 	}
 
 	glog.Infof("Deleted PV %q", pv.Name)
 
+}
+
+func (d *Deleter) deleteFilePV(pv *v1.PersistentVolume, mountPath string, config common.MountConfig) error {
+	glog.Infof("Deleting PV file volume %q contents at hostpath %q, mountpath %q", pv.Name, pv.Spec.Local.Path,
+		mountPath)
+	if err := d.VolUtil.DeleteContents(mountPath); err != nil {
+		return err
+	}
+
 	return nil
 }
 
-// cleanupBlockPV is meant to execute cleanup concurrently with other such cleanup processes
-// cmdio is a param that can suppoly io to the script being invoked. Its main purpose currently is
-// to enable better testing of the code path with output validation etc.
-func (d *Deleter) cleanupBlockPV(pv *v1.PersistentVolume, mountPath string, config common.MountConfig) error {
+func (d *Deleter) cleanupBlockPV(pv *v1.PersistentVolume, blkdevPath string, config common.MountConfig) error {
+
 	// Make absolutely sure here that we are not deleting anything outside of mounted dir
-	if !strings.HasPrefix(mountPath, config.MountDir) {
-		err := fmt.Errorf("Unexpected error pv %q mountPath %s but mount dir is %s", pv.Name, mountPath,
+	if !strings.HasPrefix(blkdevPath, config.MountDir) {
+		err := fmt.Errorf("Unexpected error pv %q mountPath %s but mount dir is %s", pv.Name, blkdevPath,
 			config.MountDir)
 		glog.Error(err)
 		return err
 	}
 
 	if len(config.BlockCleanerCommand) < 1 {
 		err := fmt.Errorf("Blockcleaner command was empty for pv %q ountPath %s but mount dir is %s", pv.Name,
-			mountPath, config.MountDir)
+			blkdevPath, config.MountDir)
 		glog.Error(err)
 		return err
 	}
 
 	cleaningInfo := fmt.Errorf("Starting cleanup of Block PV %q, this may take a while", pv.Name)
 	d.RuntimeConfig.Recorder.Eventf(pv, v1.EventTypeNormal, common.VolumeDelete, cleaningInfo.Error())
-
-	err := d.ProcTable.MarkRunning(pv.Name)
-	if err != nil {
-		return err
-	}
-	go d.asyncCleanupBlockPV(pv, mountPath, config.BlockCleanerCommand[0], config.BlockCleanerCommand[1:]...)
-
-	return nil
-}
-
-func (d *Deleter) asyncCleanupBlockPV(pv *v1.PersistentVolume, blkdevPath string, exe string, exeArgs ...string) error {
-
-	// Make sure we mark as done when we exit this function
-	defer d.ProcTable.MarkDone(pv.Name)
 	glog.Infof("Deleting PV block volume %q device hostpath %q, mountpath %q", pv.Name, pv.Spec.Local.Path,
 		blkdevPath)
 
 	glog.Infof("Starting cleanup of pv %q", pv.Name)
```

Comment:
This log msg can be removed since there's a similar one above

Reply:
done